### DISCUSSION TITLE
Previous commit SHA for the main branch

### DISCUSSION THREAD

**Discussion by brettcurtis** [community/community#52120] [Actions]
### Select Topic Area

Question

### Body

Howdy! I'm trying to get the previous commit SHA for the main branch on a workflow run on push to main. Is there a actions native way to get this:

```none
git rev-parse main~1
```

or do I need to create a step to set it?

**Comment by airtower-luna** [comment]
Do you want the commit the branch pointed at before the push, or literally `main~1`? Because those two might be very different.

If you want the former `github.event.before` should work with a push event.

**Reply by brettcurtis** [reply]
Thanks for the help! I decided what I was trying to do was a bit flakey, so I gave up. I would only sometimes compare the correct commits depending on how active the repo was. I think the `github.event.before` would have worked for me initially, but I didn't run my checkout step before testing it.
